lib/sensu/client/http_socket.rb

```diff
@@ -100,6 +108,9 @@ def send_response(status, status_string, content)
       end
 
       def process_request_info
+        if @settings[:client][:http_socket][:protect_all_endpoints]
```

Comment:
I'll go ahead and create a sensu-settings PR to add validation for `protect_all_endpoints`.